### PR TITLE
new(tests): introduce the first google benchmark

### PR DESCRIPTION
**Pull request by Andreagit97**
**What type of PR is this?**

/kind feature

**Any specific area of the project related to this PR?**

/area tests

**Does this PR require a change in the driver versions?**

**What this PR does / why we need it**:

This PR introduces the first Google benchmark. This is just the scaffolding we should add more tests to it. For example, a good candidate could be the scap-file benchmark that we are running today in our perf CI @FedeDP 

**Which issue(s) this PR fixes**:

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

**Comment by FedeDP** [comment]
Love this, thank you!
Any chance to update the CI jobs to also run benchmarks (so that for now we test that they build and work fine, but in the future we will actually use the benchmark data to generate diff from master)?

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -36,7 +36,16 @@ jobs:
     steps:
       - name: Install deps ⛓️
         run: |
-          apt update && apt install -y --no-install-recommends ca-certificates cmake build-essential git clang llvm pkg-config autoconf automake libtool libelf-dev wget libc-ares-dev libcurl4-openssl-dev libssl-dev libtbb-dev libjq-dev libjsoncpp-dev libgrpc++-dev protobuf-compiler-grpc libgtest-dev libprotobuf-dev linux-headers-${{ matrix.arch }}
+          apt update && apt install -y --no-install-recommends ca-certificates build-essential git clang llvm pkg-config autoconf automake libtool libelf-dev wget libc-ares-dev libcurl4-openssl-dev libssl-dev libtbb-dev libjq-dev libjsoncpp-dev libgrpc++-dev protobuf-compiler-grpc libgtest-dev libprotobuf-dev linux-headers-${{ matrix.arch }}
+
+      - name: Install a recent version of cmake ⛓️
+        run: |
+          curl -L -o /tmp/cmake.tar.gz https://github.com/Kitware/CMake/releases/download/v3.22.5/cmake-3.22.5-linux-$(uname -m).tar.gz
```
Comment (Andreagit97) [inline .github/workflows/ci.yml:43]: we need at least version `3.11` for  `FetchContent`

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -36,7 +36,16 @@ jobs:
     steps:
       - name: Install deps ⛓️
         run: |
-          apt update && apt install -y --no-install-recommends ca-certificates cmake build-essential git clang llvm pkg-config autoconf automake libtool libelf-dev wget libc-ares-dev libcurl4-openssl-dev libssl-dev libtbb-dev libjq-dev libjsoncpp-dev libgrpc++-dev protobuf-compiler-grpc libgtest-dev libprotobuf-dev linux-headers-${{ matrix.arch }}
+          apt update && apt install -y --no-install-recommends curl ca-certificates build-essential git clang llvm pkg-config autoconf automake libtool libelf-dev wget libc-ares-dev libcurl4-openssl-dev libssl-dev libtbb-dev libjq-dev libjsoncpp-dev libgrpc++-dev protobuf-compiler-grpc libgtest-dev libprotobuf-dev linux-headers-${{ matrix.arch }}
+
+      - name: Install a recent version of cmake ⛓️
```
Comment (FedeDP) [inline .github/workflows/ci.yml:41]: We can leverage https://github.com/marketplace/actions/actions-setup-cmake ;)

Comment (FedeDP) [reply .github/workflows/ci.yml]: Uh it does not support arm64...

**Review by FedeDP — APPROVED** [review]
/approve

**Review by jasondellaluce — APPROVED** [review]
/approve

**Pull request merged**